### PR TITLE
Add minimal BuildKubernetesNodeConfig unit test

### PR DESCRIPTION
**Pull request by ncdc**
Add minimal test for BuildKubernetesNodeConfig to ensure it sets the cloud
provider dep appropriately.

xref #11620 

cc @smarterclayton @deads2k @sttts @jsafrane

### CODE REVIEW

**Comment by ncdc** [comment]
[test]

**Comment by jsafrane** [comment]
That was fast! I would spend at least a day writing this.

IMO, you can remove `TestBuildCloudProviderNone`, as it tests `"cloud-provider": ""`. And with a little extra code you could check for an error and remove `TestBuildCloudProviderError` too.

**Comment by smarterclayton** [comment]
Thanks @ncdc

**Comment by ncdc** [comment]
@jsafrane yeah, we could definitely do that. I've got other stuff I need to move on to, so (assuming this passes) I'd recommend merging as-is and we can follow up later for a bit of cleanup. Plus, it would be useful to add more test coverage of BuildKubernetesNodeConfig, which isn't something I tried to do here.

**Comment by ncdc** [comment]
Legitimate test failure. Working on fixing.

File: pkg/cmd/server/kubernetes/node_config_test.go [hunk 1]
```diff
@@ -206,3 +213,117 @@ func TestBuildCloudProviderError(t *testing.T) {
...
+pLeyzeecPuoQsdO6SEuqhUM1ivUN4GyXIR1+aW2baMwMXpjX9VIJu5d4FqtGi6SD
+RfV+tbERWwifPJlN+ryuvqbbDxrjQeXhemeo7yrJdgJ1oyDmoM5pTiSUUmltvQID
+AQABo1AwTjAdBgNVHQ4EFgQUOVuieqGfp2wnKo7lX2fQt+Yk1C4wHwYDVR0jBBgw
+FoAUOVuieqGfp2wnKo7lX2fQt+Yk1C4wDAYDVR0TBAUwAwEB/zANBgkqhkiG9w0B
+AQsFAAOBgQA8VhmNeicRnKgXInVyYZDjL0P4WRbKJY7DkJxRMRWxikbEVHdySki6
+jegpqgJqYbzU6EiuTS2sl2bAjIK9nGUtTDt1PJIC1Evn5Q6v5ylNflpv6GxtUbCt
+bGvtpjWA4r9WASIDPFsxk/cDEEEO6iPxgMOf5MdpQC2y2MU0rzF/Gg==
+-----END CERTIFICATE-----`
+
+func writeTempFile(dir, fileName, contents string) error {
```
Comment (sttts) [inline pkg/cmd/server/kubernetes/node_config_test.go]: ioutil.WriteFile does basically the same.

File: pkg/cmd/server/kubernetes/node_config_test.go [hunk 2]
```diff
@@ -206,3 +213,117 @@ func TestBuildCloudProviderError(t *testing.T) {
...
+			},
+
+			// actual test items
+			KubeletArguments: configapi.ExtendedArguments{
+				"cloud-provider": []string{test.cloudProvider},
+			},
+		}
+		nc, err := BuildKubernetesNodeConfig(options, enableProxy, enableDNS)
+		if err != nil {
+			t.Error(err)
```
Comment (sttts) [inline pkg/cmd/server/kubernetes/node_config_test.go]: continue

Comment (sttts) [inline pkg/cmd/server/kubernetes/node_config_test.go]: or else below

File: pkg/cmd/server/kubernetes/node_config_test.go [hunk 3]
```diff
@@ -206,3 +213,117 @@ func TestBuildCloudProviderError(t *testing.T) {
...
+		}
+		if test.expectProvider {
+			if nc.KubeletDeps.Cloud == nil {
+				t.Errorf("%s: expected cloud provider to be set", test.name)
```
Comment (sttts) [inline pkg/cmd/server/kubernetes/node_config_test.go]: ... to be set, but found nil

**Comment by ncdc** [comment]
[test] #10988

**Comment by ncdc** [comment]
[test] #11682

**Comment by ncdc** [comment]
[test]

**Comment by ncdc** [comment]
#10487 [test]

**Comment by ncdc** [comment]
FYI @liggitt

**Pull request closed without merging**